### PR TITLE
Unroll shader loop when running on Windows to avoid shader crashes

### PR DESCRIPTION
**Pull request by daniel-wer**
Review hiding whitespace changes!

I added that both the fragment and the vertex shaders are unit-tested.

### URL of deployed dev instance (used for testing):
- https://usedifferentshaderforwindows.webknossos.xyz

### Steps to test:
- Enable interpolation and open a couple of dataset on Windows

### TODOs:
- [ ] ...

### Issues:
- fixes #

------
(Please delete unneeded items, merge only when none are left open)
- [x] Added changelog entry (create a `$PR_NUMBER.md` file in `unreleased_changes` or use `./tools/create-changelog-entry.py`)
- [ ] Added migration guide entry if applicable (edit the same file as for the changelog)
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: frontend/javascripts/test/shaders/shader_syntax.spec.ts [hunk 1]
```diff
@@ -1,277 +1,312 @@
...
 describe("Shader syntax", () => {
   const originalWarn = console.warn;
 
   interface TestContext {
-    warningEmittedCount: number;
+    warnings: { emittedCount: number };
   }
 
-  beforeEach<TestContext>(async (context) => {
+  beforeEach<TestContext>(async ({ warnings }) => {
```
Comment (daniel-wer) [inline frontend/javascripts/test/shaders/shader_syntax.spec.ts]: Due to the way the `it.extend` method and fixtures work in vitest, this argument had to be destructured which is why I had to introduce the inner object to be able to increase the number of emitted warnings in the context object.

File: frontend/javascripts/test/shaders/shader_syntax.spec.ts [hunk 2]
```diff
@@ -1,277 +1,312 @@
...
-  });
+        orderedColorLayerNames: ["color_layer_1", "color_layer_2"],
+        segmentationLayerNames: [],
+        magnificationsCount: mags.length,
+        voxelSizeFactor: [1, 1, 1],
+        isOrthogonal: true,
+        voxelSizeFactorInverted: [1, 1, 1],
+        useInterpolation: false,
+        tpsTransformPerLayer: {},
+        isWindows: false,
```
Comment (philippotto) [inline frontend/javascripts/test/shaders/shader_syntax.spec.ts:60]: would it make sense to use vitests `describe.each` so that all existing tests can be tested with isWindows true and false?

Comment (daniel-wer) [reply frontend/javascripts/test/shaders/shader_syntax.spec.ts:60]: Ah I didn't realize I could introduce the `describe.{each,for}` at the top-level. I could use that for the fragment/vertex shader parameterization to avoid some code duplication. However, I'm not sure we want to execute each test with isWindows true and false as the same argument could be made for all the other flags (useInterpolation, isOrthogonal, ...) and we would end up executing a loooot of variations, wouldn't we?

Comment (philippotto) [reply frontend/javascripts/test/shaders/shader_syntax.spec.ts:60]: yes, that's true. the code in question doesn't have much syntactic variability. so, I guess testing isWindows true and false is probably enough :)
